dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/dxf2/sync/TrackerSynchronizationTest.java

```diff
@@ -85,15 +95,21 @@ private void prepareDataForTest()
         manager.save( tet );
         OrganisationUnit ou = createOrganisationUnit( 'a' );
         manager.save( ou );
-        TrackedEntityInstance tei = createTrackedEntityInstance( 'a', ou, teaA );
-        tei.setTrackedEntityType( tet );
-        TrackedEntityAttributeValue teavB = createTrackedEntityAttributeValue( 'b', tei, teaB );
-        TrackedEntityAttributeValue teavA = createTrackedEntityAttributeValue( 'a', tei, teaA );
-        manager.save( tei );
+        TrackedEntityInstance teiWithAttributes = createTrackedEntityInstance( 'a', ou, teaA );
+        teiWithAttributes.setTrackedEntityType( tet );
+        teiWithAttributes.setUid( TEI_UID );
+        TrackedEntityAttributeValue teavB = createTrackedEntityAttributeValue( 'b', teiWithAttributes, teaB );
+        TrackedEntityAttributeValue teavA = createTrackedEntityAttributeValue( 'a', teiWithAttributes, teaA );
+        manager.save( teiWithAttributes );
         trackedEntityAttributeValueService.addTrackedEntityAttributeValue( teavA );
         trackedEntityAttributeValueService.addTrackedEntityAttributeValue( teavB );
-        tei.getTrackedEntityAttributeValues().addAll( List.of( teavA, teavB ) );
-        manager.update( tei );
+        teiWithAttributes.getTrackedEntityAttributeValues().addAll( List.of( teavA, teavB ) );
+        manager.update( teiWithAttributes );
+        TrackedEntityInstance teiA = createTrackedEntityInstance( 'b', ou );
+        teiA.setTrackedEntityType( tet );
+        teiA.setLastSynchronized( TOMORROW );
```

Comment:
Its confusing that the `lastSynchronized` is in the future 🤔 It might not matter in terms of testing that the query returns the expected results though.

I think it will be clearer if we set the `lastUpdated` and `lastSynchronized` here. I am guessing `lastUpdated` is supposed to be < `lastSynchronized` for `teiA`.